osquery/config/config.cpp

```diff
@@ -530,6 +553,58 @@ void stripConfigComments(std::string& json) {
   json = sink;
 }
 
+Expected<ConfigMap, Config::RestoreConfigError> Config::restoreConfigBackup() {
+  if (!FLAGS_config_enable_backup) {
```

Comment:
This is more of a personal preference but I would prefer this was checked on the logic that calls this function. In general I think it's better to separate the implementation of a functionality from the code that checks whether that functionality should be executed.